### PR TITLE
feat: Account onboarding using CFN template link

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:

This PR enables two features for AppStream
1. Onboarding a SWB member account onto a preexisting AWS account
* The Add AWS Account screen now includes text boxes for AppStream configs.
AddAccount
2. Upgrading a preexisting SWB member account to support AppStream
* When the user clicks the "Re-onboard Account" button, we check if that account has any active workspaces. If they have active workspaces we give them a warning.
warningMessage

* If there are no active workspaces for that account, we'll takes them to a screen which allows them to enter AppStream config
UpdateAwsAccount

The final screen for both features is the Update Cloudformation screen. This screen gives the customer step by step instructions on how to launch the new CF template in their member account.
OnboardAccount

*Note:* Upgrading a preexisting member account before Aidan's changes, involves creating a new CF stack in the member account. DDB Aws Accounts table will be updated to point to this new CF stack. The old CF stack should have no active workspaces as those workspaces must be deleted before SWB allows them to upgrading their account to AppStream enabled.

**Merging Strategy**: This PR only contains code changes that I've made since merging `feat-update-user-permissions` (Aidan's onboarding work) and the code in `feat-secure-workspace-egress`. I've merged the two branches together to create the new branch  `merged-workspace-egress-update-user-permissions`. This PR contains my changes on top of those two merged branches. 

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [x] If new dependencies have been added, have they been pinned to specific versions?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ahl27** [comment]
LGTM from the APU side but I'll let someone from AppStream have the final approval. Thanks for getting this merged Tim, looks great.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccount.js
Comment (SanketD92) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccount.js]: For other operations like provisioning environments, we determine if an account is AppStream-configured or not by checking if at least these attributes are available: appStreamFleetName, appStreamSecurityGroupId, appStreamStackName.

Let's keep that determination consistent if possible, like:
```
self.isAppStreamConfigured = !_.isUndefined(rawAwsAccounts.appStreamStackName) && !_.isUndefined(rawAwsAccounts.appStreamFleetName) && !_.isUndefined(rawAwsAccounts.appStreamSecurityGroupId);
```

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-config-vars-service.js
Comment (SanketD92) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-config-vars-service.js]: Doesn't need to be an array, right?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AwsAccountsList.js
Comment (SanketD92) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AwsAccountsList.js]: Could we add getter methods for each of these stores, and decorate them as `computed`? This would allow us to clean up the `Promise.all()` type initialization below.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/forms/AddUpdateAwsAccountForm.js
Comment (SanketD92) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/models/forms/AddUpdateAwsAccountForm.js:36]: Could we enter default values in this form?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AwsAccountsList.js
Comment (nguyen102) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AwsAccountsList.js]: Note, I'm moving this logic to check for active account every time a user clicks the `Re-Onboard Account` button. This is to prevent the scenario where a new workspace is spun up before the user decides to click the `Re-Onboard Account` button.

On that note, if we used `computed`, will `mobx` cache the API results? We want live data and not any cached results.

Info [here](https://mobx.js.org/computeds.html)
`Computed values can be used to derive information from other observables. They evaluate lazily, caching their output and only recomputing if one of the underlying observables has changed.`

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/forms/AddUpdateAwsAccountForm.js
Comment (nguyen102) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/models/forms/AddUpdateAwsAccountForm.js:36]: Hmm, we could do that, but then users won't be able to see the description of the field in the placeholder value. I'm leaning towards not having a default value so they can be informed of their decision and select the values that best suit their need. Thoughts? 
Screen Shot 2021-07-28 at 5 04 05 PM

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-config-vars-service.js
Comment (nguyen102) [reply addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-config-vars-service.js]: updated

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccount.js
Comment (nguyen102) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccount.js]: updated

**Comment by jn1119** [comment]
>  I've merged the two branches together to create the new branch `merged-workspace-egress-update-user-permissions`. This PR contains my changes on top of those two merged branches.

Can we merge the change in `feat-secure-workspace-egress` itself instead of creating a new branch? Or are you proposing that all the egress and update user changes are made to this new branch until we merge in develop?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AccountCard.js
Comment (jn1119) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AccountCard.js:267]: Seems like an error, no?

**Review by jn1119 — APPROVED** [review]

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AccountCard.js
Comment (jn1119) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AccountCard.js:267]: Discussed offline. From UX point of view warning seems better.

**Pull request merged**
